python/ql/src/experimental/Security/CWE-208/TimingAttackAgainstHeaderValue/TimingAttackAgainstHeaderValue.ql

```diff
@@ -0,0 +1,33 @@
+/**
+ * @name Timing attack against header value
+ * @description Use of a non-constant-time verification routine to check the value of an HTTP header,
+ *              possibly allowing a timing attack to infer the header's expected value.
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id py/timing-attack-against-header-value
+ * @tags security
+ *       external/cwe/cwe-208
+ */
+
+import python
+import semmle.python.dataflow.new.DataFlow
+import semmle.python.dataflow.new.TaintTracking
+import experimental.semmle.python.security.TimingAttack
+import DataFlow::PathGraph
+
+/**
+ * A configuration tracing flow from  a client Secret obtained by an HTTP header to a unsafe Comparison.
+ */
+class ClientSuppliedSecretConfig extends TaintTracking::Configuration {
+  ClientSuppliedSecretConfig() { this = "ClientSuppliedSecretConfig" }
+
+  override predicate isSource(DataFlow::Node source) { source instanceof ClientSuppliedSecret }
+
+  override predicate isSink(DataFlow::Node sink) { sink instanceof CompareSink }
+}
+
+from ClientSuppliedSecretConfig config, DataFlow::PathNode source, DataFlow::PathNode sink
+where config.hasFlowPath(source, sink) and not sink.getNode().(CompareSink).flowtolen()
```

Comment:
This `flowtolen` guard is a very strange way of addressing https://github.com/github/securitylab/issues/691#issuecomment-1443208643. I'm assuming that in the cases you were looking at, these `len` calls were always present? What would happen if there isn't any call to `len`? In that case you would still want `in` comparisons to not be considered.

I think it would be better to just remove `in` as a potential sink entirely.